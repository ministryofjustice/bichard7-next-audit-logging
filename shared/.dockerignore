coverage
dist
node_modules
test-results
.dockerignore
jest.config.js
README.md
