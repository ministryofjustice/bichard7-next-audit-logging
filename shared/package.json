{
  "name": "shared",
  "description": "Hosts common types and services",
  "version": "0.0.1",
  "private": true,
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist/**/*"
  ],
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^4.9.0",
    "@typescript-eslint/parser": "^4.9.0",
    "eslint": "^7.15.0",
    "jest": "^26.6.3",
    "jest-junit": "^12.0.0",
    "ts-jest": "^26.4.4",
    "typescript": "^4.1.2"
  },
  "dependencies": {
    "aws-sdk": "^2.870.0"
  },
  "scripts": {
    "build": "tsc -p tsconfig.json",
    "build:dev": "npm run build",
    "setup:env": "cd .. && ./environment/setup.sh && cd -",
    "test": "npm run setup:env && ../scripts/test.sh",
    "test:unit": "../scripts/test.sh unit",
    "test:integration": "npm run setup:env && ../scripts/test.sh integration",
    "test:e2e": "npm run setup:env && ../scripts/test.sh e2e",
    "test:ci": "../scripts/test.sh ci"
  }
}
