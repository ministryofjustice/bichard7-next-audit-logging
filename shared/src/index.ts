export * from "./DynamoGateway"
export * from "./types"
export * from "./utils"
