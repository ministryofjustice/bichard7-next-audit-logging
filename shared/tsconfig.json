{
  "extends": "../.config/tsconfig.json",
  "compilerOptions": {
    "strictNullChecks": false,
    "outDir": "dist",
    "types": ["node", "jest"],
    "sourceMap": false,
    "baseUrl": ".",
    "preserveSymlinks": false,
    "declaration": true,
    "paths": {
      "src/*": ["src/*"]
    }
  },
  "compileOnSave": true,
  "include": ["src/**/*.ts"]
}
