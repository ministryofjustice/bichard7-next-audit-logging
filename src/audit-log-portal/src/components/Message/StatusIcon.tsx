import { Tooltip } from '@material-core/ui'
import type { AuditLog } from "shared-types"
import getStatusIcon from "./getStatusIcon"

interface Props {
  message: AuditLog
}

const StatusIcon = ({ message }: Props) => {
  const Icon = getStatusIcon(message.status)

  return (
    <Tooltip title={message.lastEventType || message.status} placement="top">
      <div>
        <Icon />
      </div>
    </Tooltip>
  )
}

export default StatusIcon
