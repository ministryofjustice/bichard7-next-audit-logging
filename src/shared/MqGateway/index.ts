export { default as createMqConfig } from "./createMqConfig"
export { default as MqGateway } from "./MqGateway"
export { default as TestMqGateway } from "./TestMqGateway"
