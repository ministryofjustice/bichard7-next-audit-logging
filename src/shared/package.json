{
  "name": "shared",
  "description": "Hosts common types and services",
  "version": "0.0.1",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "files": [
    "build/**/*"
  ],
  "devDependencies": {
    "@types/jest": "^29.1.2",
    "@types/pg": "^8.6.5",
    "@types/pino": "^6.0.0",
    "@types/stompit": "^0.26.3",
    "@types/uuid": "^8.3.4",
    "@types/xml2js": "^0.4.11",
    "@typescript-eslint/eslint-plugin": "^5.39.0",
    "@typescript-eslint/parser": "^5.39.0",
    "eslint": "^8.25.0",
    "jest": "^29.1.2",
    "jest-junit": "^14.0.1",
    "rimraf": "^3.0.2",
    "shared-testing": "file:../shared-testing",
    "ts-jest": "^29.0.3",
    "typescript": "^4.8.4"
  },
  "dependencies": {
    "@types/lodash.maxby": "^4.6.7",
    "@types/lodash.minby": "^4.6.7",
    "aws-sdk": "^2.1231.0",
    "axios": "^1.1.2",
    "date-fns": "^2.29.3",
    "lodash.maxby": "^4.6.0",
    "lodash.minby": "^4.6.0",
    "pg": "^8.8.0",
    "pg-native": "^3.0.1",
    "pino": "^6.13.0",
    "shared-types": "file:../shared-types",
    "stompit": "^1.0.0",
    "uuid": "^9.0.0",
    "xml2js": "^0.4.23",
    "zlib": "^1.0.5"
  },
  "scripts": {
    "clean": "rimraf build",
    "build": "npm run clean && tsc -p tsconfig.json",
    "build:prod": "npm run clean && tsc -p tsconfig.prod.json",
    "test": "jest --runInBand",
    "test:unit": "jest --testPathIgnorePatterns='integration'",
    "test:unit:watch": "jest --watchAll --testPathIgnorePatterns='integration'",
    "test:integration": "jest --testPathPattern=integration"
  }
}
