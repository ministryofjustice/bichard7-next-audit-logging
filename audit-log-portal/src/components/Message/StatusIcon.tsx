import { Tooltip } from "@material-ui/core"
import { AuditLog } from "shared"
import getStatusIcon from "./getStatusIcon"

interface Props {
  message: AuditLog
}

const StatusIcon = ({ message }: Props) => {
  const Icon = getStatusIcon(message.status)

  return (
    <Tooltip title={message.lastEventType || message.status} placement="top">
      <div>
        <Icon />
      </div>
    </Tooltip>
  )
}

export default StatusIcon
