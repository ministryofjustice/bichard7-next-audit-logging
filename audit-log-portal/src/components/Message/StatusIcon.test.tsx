import { render } from "testing/render"
import { AuditLog } from "shared"
import StatusIcon from "./StatusIcon"

test.each<string>(["Processing", "Completed", "Error message"])("matches the snapshot when status is %s", (status) => {
  const message = new AuditLog("ExternalCorrelationId", new Date(), "XML")
  message.status = status

  const { container } = render(<StatusIcon message={message} />)

  expect(container).toMatchSnapshot()
})
