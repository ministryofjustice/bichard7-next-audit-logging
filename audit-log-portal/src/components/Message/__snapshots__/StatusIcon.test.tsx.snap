// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when status is Completed 1`] = `
<div>
  <div
    class=""
    title="Completed"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check-circle fa-w-16 SuccessIcon__ColoredIcon-j6z8ag-0 jyZfvZ"
      data-icon="check-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`matches the snapshot when status is Error 1`] = `
<div>
  <div
    class=""
    title="Error"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-exclamation-circle fa-w-16 ErrorIcon__ColoredIcon-sc-1n9o5bs-0 jMrgmb"
      data-icon="exclamation-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`matches the snapshot when status is Processing 1`] = `
<div>
  <div
    class=""
    title="Processing"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
